pyiron_base/dataclasses/job.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class Server:
```

Comment:
Ah, we're not packaging `pyiron_io`, yet right? Otherwise would be good to import directly.

Reply:
Not yet. At the moment I simply copied the dataclass. It is also not clear to me how to handle the versioning of the dataclasses efficiently, for example we might want to have multiple version of the `Server` dataclass in `pyiron_io` for backwards compatibility while `pyiron_base` should use only the latest one. In addition, a number of dataclasses in `pyiron_io` are very specific to atomistic simulation, so we might want to separate those from the general dataclasses for the job management and so on. Still once we have a central repository it should be quick to update the imports here.

Reply:
For the versioning we might set up a simple dictionary in pyiron_io 
```python
@dataclass
class ServerVersion0_9_12:
  ...
@dataclass
class Server:
  ...
ServerClasses = {
  '0.9.2': ServerVersion0_9_12,
  'latest': Server
}
```
then reading code could grab the correct one
```python
import pyiron_base
ServerClasses.get(pyiron_base.__version__, Server)
```
or so, but am totally fine with delaying this.